sympy/ntheory/residue_ntheory.py

```diff
@@ -742,6 +742,63 @@ def _nthroot_mod1(s, q, p, all_roots):
         return res
     return min(res)
 
+def _nthroot_mod_composite(a, n, p):
+    """
+    Find the solutions to ``x**n = a mod p`` when p is not prime.
+    """
+    f = factorint(p)
+    dd = {}
+    for b, e in f.items():
+        tot_roots = set()
+        if e == 1:
+            root = nthroot_mod(a, n, b, True)
+            if root == None:
+                root = []
+            if (pow(b, n) - a) % b == 0:
+                root.append(b)
+            for num in root:
+                tot_roots.add(num)
+        else:
+            roots = nthroot_mod(a, n, b, True)
+            if roots == None:
+                roots = []
+            if (pow(b, n) - a) % b == 0:
+                roots.append(b)
+            for i in roots:
+                root = i
+                diff = (n * pow(i, n-1)) % b
+                if diff != 0:
+                    for j in range(1, e):
+                        root = (root - (pow(root, n) - a)* mod_inverse(diff, b)) % pow(b, j+1)
+                    tot_roots.add(root)
+                else:
+                    new_base = b
+                    roots_in_base = set({i})
+                    while new_base < pow(b, e):
+                        new_base = new_base * b
+                        new_roots = set()
+                        for k in roots_in_base:
+                            if (pow(k, n) - a) % (new_base) != 0:
+                                continue
+                            while(k not in new_roots):
+                                new_roots.add(k)
+                                k = (k + (new_base // b)) % new_base
+                        roots_in_base = (new_roots)
+                    tot_roots = tot_roots | roots_in_base
+        dd[pow(b, e)] = tot_roots
+    a=[]
+    m=[]
+    from sympy.ntheory.modular import crt
+    for x, y in dd.items():
+        m.append(x)
+        a.append(list(y))
+    import itertools
```

Comment:
`return sorted(set(crt(m, list(i))[0] for i in cartes(*a)))` and import `cartes` from utilities.iterables at the top of the file